### ISSUE TITLE
MaskGAN code contains several bugs and is incomplete

### ISSUE THREAD

**Issue by schmiflo** [tensorflow/models#6818]
The maskgan implementation does not meet the code quality usually found in the tensorflow codebase. It contains several bugs and many shortcomings.

## bugs
- In  models/research/maskgan/data/ptb_loader.py 
`` return f.read().decode("utf-8").replace("\n", "<eos>").split()``
attaches the `<eos>` to random symbols.

- In models/research/maskgan/generate_samples.py 
`tf.Supervisor()`
does not exist in the recommended TF version 1.5. It was probably always under tf.train

## shortcomings

- Corpus filenames i.e. `ptb.train.txt` are hardcoded.

- It is very unclear how the code in `generate_samples.py` can generate *unconditional* samples. It iterates across the dataset and the output -- even masking all tokens -- impacts the generated output (as tested by providing a dummy corpus).

- Essential documentation, for example for pretraining the LM, is marked as WIP for over a year.
- Many features, for example generator pre-training, raise NotImplemented exceptions. 

### System information
- **What is the top-level directory of the model you are using**:
models/research/maskgan/
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
ubuntu 18.04
- **TensorFlow installed from (source or binary)**:
binary
- **TensorFlow version (use command below)**:
1.5.1 as is recommended but the code can be found in the current master
- **Bazel version (if compiling from source)**:
n/a
- **CUDA/cuDNN version**:
n/a
- **GPU model and memory**:
n/a
- **Exact command to reproduce**:

**Comment by a-dai** [comment]
@liamb315 Could you look at this?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
